lib/datadog/tracing/sampling/rate_sampler.rb

```diff
@@ -20,6 +20,9 @@ class RateSampler < Sampler
         # * +sample_rate+: the sample rate as a {Float} between 0.0 and 1.0. 0.0
         #   means that no trace will be sampled; 1.0 means that all traces will be
         #   sampled.
+        #
+        # DEV-2.0: Allow for `sample_rate` zero (drop all) to be allowed. This eases
+        # DEV-2.0: usage for many consumers of the {RateSampler} class.
```

Comment:
Interesting. Can you clarify?

Reply:
I expanded the comment in the PR, but here's the gist:
All internal users of `RateSampler` (RuleSampler and now Single Span Sampling) want sample_rate == 0 to mean "drop all", but they can't do that because of the validation that happens in the `RateSampler` initializer.

The way they get around it is to not set the value in the initializer, but call:
```ruby
sampler = RateSampler.new
sampler.sample_rate = sample_rate # There's no validation here
```
This bypasses the validation. Ideally, the `RateSampler` would respect any rate between 0.0 and 1.0.